parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -122,6 +134,17 @@ ensureRPC() {
     systemctlEnableAndStart rpc-statd || exit $ERR_SYSTEMCTL_START_FAIL
 }
 
+ensureAuditD() {
+  if [[ "${AUDITD_ENABLED}" == true ]]; then
+    systemctlEnableAndStart auditd || exit $ERR_SYSTEMCTL_START_FAIL
+  else
+    apt list --installed | grep 'auditd'
+    if [ $? -eq 0 ]; then
+      systemctlDisableAndStop auditd || exit $ERR_SYSTEMCTL_START_FAIL
```

Comment:
why not `apt-get purge` if it's not enabled?

Reply:
Just trying to avoid more apt operations at runtime for all cluster creates. Strictly speaking it's not doing anything except taking up space on the filesystem

Reply:
this seems like it's introducing more failure potential since with the existing cleanup functions we simply attempt to purge but without returning a non-zero exit code if the cleanup is unsuccessful (cleaning up an unnecessary package is arguably not critical to cluster deployment success). This breaks that pattern by saying: "return $ERR_SYSTEMCTL_START_FAIL  if we can't disable and stop the service" which could fail for reason xyz.

Reply:
It's very important that we (1) ensure that auditd isn't running and (2) that it won't start after reboot, so we definitely need this validation.

Reply:
You were right all along:

https://github.com/Azure/aks-engine/pull/2078

:)